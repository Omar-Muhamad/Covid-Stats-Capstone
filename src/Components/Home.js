import { useDispatch, useSelector } from 'react-redux';
import { increament, decreament } from '../Redux/counter/counter';

const Home = () => {
  const counter = useSelector((state) => state.counter);
  const dispatch = useDispatch();
  return (
    <section className="hero">
      <h1>React Redux Template</h1>
      <button type="button" onClick={() => dispatch(increament())}>+</button>
      <p>{counter}</p>
      <button type="button" onClick={() => dispatch(decreament())}>-</button>
    </section>
  );
};

export default Home;
