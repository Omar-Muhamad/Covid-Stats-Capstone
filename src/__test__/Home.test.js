import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import store from '../Redux/configreStore';
import Home from '../Components/Home';

describe('Test Home render', () => {
  test('Test Home Render', () => {
    const tree = render(
      <Provider store={store}>
        <Home />
      </Provider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
