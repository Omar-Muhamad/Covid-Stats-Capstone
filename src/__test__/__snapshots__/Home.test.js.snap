// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Home render Test Home Render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="absolute z-10 w-screen h-screen bg-[#5787E5] flex flex-col gap-4 justify-center items-center"
      >
        <div
          class="w-[110px] h-[110px] border-[16px] border-t-[#5787E5] rounded-full border-white animate-spin"
        />
        <div
          class="text-white text-2xl font-bold"
        >
          Loading...
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="absolute z-10 w-screen h-screen bg-[#5787E5] flex flex-col gap-4 justify-center items-center"
    >
      <div
        class="w-[110px] h-[110px] border-[16px] border-t-[#5787E5] rounded-full border-white animate-spin"
      />
      <div
        class="text-white text-2xl font-bold"
      >
        Loading...
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
